### ISSUE TITLE
Implementation of Memory Buffer

### ISSUE THREAD

**Issue by KyleZheng1997** [google-research/simclr#115]
Hello, I'm just wondering how do you implement the memory buffer in SimCLRV2?
Did you follow exactly the same implementation with MoCo ? (i.e. use the output of the target encoder as the positive embedding and treat the whole memory buffer as the negative embedding).

I'm currently trying to reproduce this part, but I found that with 100 epochs of training, I can get 66+% without memory buffer but only 60% with the memory buffer, which is quite odd.

Would you like to release the code with memory buffer or simply illustrate how you actually implement this part?
I will be really appreciated it !!

BTW, Thank you so much for the amazing work !!

**Comment by chentingpc** [comment]
Hi, the contrasting examples are from momentum encoder from current and past batches, which is consistent with MoCo.

Since the gain is small (~1%), and it would make code more complicate, so we decided not to opensource the code in this repo.

**Comment by KyleZheng1997** [comment]
How about augmentation? I noticed that MoCov2 adopts a weaker augmentation (i.e the lower bound of the random crop is 0.2, color jitter strength is 0.5). For SimCLRV2, did you still use the same augmentation with SimCLRV1?  Moreover, do you remember the result of 100 epochs of training? It will be really helpful !!

Thank you so much !!!!

**Comment by chentingpc** [comment]
Yes, the augmentation is the same as SimCLRv1 (except for brightness see [here](https://github.com/google-research/simclr/blob/3ad6700c1b139ee18e43f73546b7263a710de699/data_util.py#L41)). Weaker augmentation may help ResNet-50 but could be worse for bigger networks.

I don't have 100 epoch result at hand for exact SimCLRv2 setting, but you could refer to Table D.3 in https://arxiv.org/pdf/2011.02803.pdf, which is almost the same SimCLRv2 setting except that it doesn't use memory component and small change of hparms with lr=0.075 x sqrt(bsz) and temperature=0.2.

**Comment by KyleZheng1997** [comment]
Would you like to share at least some pseudo-code for this part? I'm still quite struggling for reproducing the performance with memory buffer. Thank you!
